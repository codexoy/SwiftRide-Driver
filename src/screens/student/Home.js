import React, { Fragment } from "react"
import {
	ScrollView,
	View,
	Text,
	Image,
	StyleSheet,
	TouchableHighlight,
	TouchableOpacity
} from "react-native"
import { connect } from "react-redux"
import ShadowRect from "../../components/ShadowRect"
import { strings } from "../../i18n"
import { Icon } from "react-native-elements"
import LessonPopup from "../../components/LessonPopup"
import { getPayments } from "../../actions/lessons"
import { MAIN_PADDING } from "../../consts"
import StudentPayments from "../../components/StudentPayments"
import StudentNextLessonView from "../../components/StudentNextLessonView"

export class Home extends React.Component {
	static navigationOptions = () => {
		return {
			title: "home",
			tabBarLabel: strings("tabs.home"),
			tabBarAccessibilityLabel: strings("tabs.home"),
			tabBarTestID: "HomeTab"
		}
	}

	constructor(props) {
		super(props)
		this.state = {
			lesson: "",
			payments: [],
			lessonPopupVisible: false,
			loading: true
		}

		this._handleRequests()
	}

	_handleRequests = async () => {
		await this._getLesson()
		await this._getPayments()
		this.setState({ loading: false })
	}
	_navigateToSettings = () => {
		this.props.navigation.navigate("Settings")
	}

	_getLesson = async () => {
		const now = new Date().toISOString()
		const resp = await this.props.fetchService.fetch(
			"/lessons/?limit=1&is_approved=true&date=ge:" + now,
			{ method: "GET" }
		)
		if (resp.json["data"].length == 0) return
		this.setState({
			lesson: resp.json["data"][0]
		})
	}

	_getPayments = async () => {
		const payments = await getPayments(this.props.fetchService)
		this.setState({
			payments: payments.payments
		})
	}

	lessonPress = () => {
		this.setState({ lessonPopupVisible: !this.state.lessonPopupVisible })
	}

	renderLesson = () => {
		const { lesson } = this.state
		return (
			<Fragment>
				<StudentNextLessonView
					onPress={() => this.lessonPress()}
					testID="lessonRowTouchable"
					lesson={lesson}
					loading={this.state.loading}
				/>
				<LessonPopup
					visible={this.state.lessonPopupVisible}
					item={lesson}
					onPress={this.lessonPress}
					testID="lessonPopup"
					navigation={this.props.navigation}
				/>
			</Fragment>
		)
	}

	render() {
		return (
			<ScrollView>
				<View style={styles.container}>
					<View style={styles.settingsIcon}>
						<TouchableOpacity
							onPress={this._navigateToSettings.bind(this)}
						>
							<Icon name="settings" type="material" size={24} />
						</TouchableOpacity>
					</View>
					<View testID="welcomeHeader" style={styles.welcomeHeader}>
						<Image
							style={styles.profilePic}
							source={{
								uri:
									"https://images.unsplash.com/photo-1535643302794-19c3804b874b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2134&q=80"
							}}
						/>
						<Text style={styles.welcomeText}>
							{strings("teacher.home.welcome", {
								name: this.props.user["name"]
							})}
						</Text>
					</View>
					<ShadowRect style={styles.schedule}>
						<Text style={styles.rectTitle} testID="schedule">
							{strings("teacher.home.next_lesson")}
						</Text>
						{this.renderLesson()}
					</ShadowRect>

					<View style={styles.fullScheduleView}>
						<TouchableHighlight
							underlayColor="lightgray"
							onPress={() => {
								this.props.navigation.navigate("Schedule")
							}}
						>
							<Text style={styles.fullSchedule}>
								{strings("teacher.home.full_schedule")}
							</Text>
						</TouchableHighlight>
						<Icon
							size={20}
							color="rgb(12, 116, 244)"
							name="ios-arrow-dropleft-circle"
							type="ionicon"
						/>
					</View>
					<ShadowRect>
						<View style={{ flex: 1, flexDirection: "row" }}>
							<Text
								testID="monthlyAmount"
								style={styles.rectTitle}
							>
								{strings("student.home.payments")}
							</Text>
							<View
								style={{
									flex: 1,
									alignItems: "flex-end",
									marginRight: "auto"
								}}
							>
								<Icon
									name="arrow-back"
									type="material"
									size={20}
								/>
							</View>
						</View>
						<StudentPayments
							sum={this.props.user.balance}
							payments={this.state.payments.slice(0, 2)}
							loading={this.state.loading}
						/>
					</ShadowRect>
				</View>
			</ScrollView>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		marginTop: 20,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center"
	},
	settingsIcon: {
		position: "absolute",
		right: MAIN_PADDING
	},
	profilePic: {
		width: 44,
		height: 44,
		borderRadius: 22,
		marginBottom: 16
	},
	welcomeHeader: {
		alignSelf: "center",
		alignItems: "center",
		marginBottom: 20
	},
	welcomeText: {
		fontFamily: "Assistant-Light",
		fontSize: 24
	},
	hoursStyle: { marginBottom: -8 },
	rectTitle: {
		fontSize: 16,
		fontWeight: "bold",
		color: "rgb(121, 121, 121)",
		alignSelf: "flex-start"
	},
	lessonRow: {
		marginTop: 12
	},
	fullScheduleView: {
		flexDirection: "row",
		flex: 1,
		alignSelf: "center",
		alignItems: "center",
		marginTop: 4,
		marginBottom: 24
	},
	fullSchedule: {
		color: "rgb(12, 116, 244)",
		marginTop: -4,
		fontWeight: "bold",
		marginRight: 8
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Home)
