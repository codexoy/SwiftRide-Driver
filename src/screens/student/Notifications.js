import React, { Fragment } from "react"
import {
	View,
	Text,
	TouchableHighlight,
	StyleSheet,
	FlatList,
	TouchableOpacity
} from "react-native"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import PageTitle from "../../components/PageTitle"
import { MAIN_PADDING, NAME_LENGTH } from "../../consts"
import { Dropdown } from "react-native-material-dropdown"
import EmptyState from "../../components/EmptyState"
import StudentsLoader from "../../components/StudentsLoader"
import Notification from "../../components/Notification"
import moment from "moment"
import LessonPopup from "../../components/LessonPopup"
import Hours from "../../components/Hours"
import ShowReceipt from "../../components/ShowReceipt"

export class Notifications extends React.Component {
	static navigationOptions = () => {
		return {
			title: "notifications",
			tabBarLabel: strings("tabs.notifications_title"),
			tabBarAccessibilityLabel: strings("tabs.notifications_title"),
			tabBarTestID: "NotificationsTab"
		}
	}
	constructor(props) {
		super(props)
		this.filterOptions = [
			{
				value: "lessons/",
				label: strings("notifications.scheduled_lessons")
			},
			{
				value: "lessons/payments",
				label: strings("notifications.payments")
			}
		]
		this.state = {
			items: [],
			page: 1,
			nextUrl: "",
			filter: "",
			refreshing: false,
			visible: []
		}

		this._dropdownChange = this._dropdownChange.bind(this)
	}

	_onNavigationFocus = () => {
		this.setState(
			{
				filter:
					this.props.navigation.getParam("filter") ||
					this.filterOptions[0]["value"],
				loading: true,
				page: 1,
				nextUrl: ""
			},
			() => {
				this._getItems()
			}
		)
	}

	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._onNavigationFocus()
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	onRefresh = () => {
		this.setState({ page: 1, nextUrl: "", refreshing: true }, () => {
			this._getItems()
		})
	}

	_getItems = async (append = false) => {
		const resp = await this.props.fetchService.fetch(
			"/" +
				this.state.filter +
				"?limit=10&order_by=created_at desc&is_approved=true&creator_id=" +
				this.props.user.my_teacher.user.id +
				"&page=" +
				this.state.page,
			{
				method: "GET"
			}
		)
		let newValue = resp.json.data
		if (append) {
			newValue = [...this.state.items, ...newValue]
		}
		this.setState({
			items: newValue,
			nextUrl: resp.json["next_url"],
			loading: false,
			refreshing: false
		})
	}

	renderPayment = ({ item, index }) => {
		const date = moment
			.utc(item.created_at)
			.local()
			.format("DD.MM.YY, HH:mm")
		return (
			<Notification
				style={styles.notification}
				key={`payment${item.id}`}
				leftSide={<Text style={styles.amount}>{item.amount}â‚ª</Text>}
				basic={
					<Fragment>
						<Text style={styles.basic}>{date}</Text>
						<ShowReceipt
							item={item}
							dispatch={this.props.dispatch}
							user={this.props.user}
							style={{ alignSelf: "flex-start" }}
						/>
					</Fragment>
				}
			/>
		)
	}

	lessonPress = item => {
		let newVisible
		if (this.state.visible.includes(item.id)) {
			// we pop it
			newVisible = this.state.visible.filter((v, i) => v != item.id)
		} else {
			newVisible = [...this.state.visible, item.id]
		}
		this.setState({ visible: newVisible })
	}

	renderLesson = ({ item, index }) => {
		const visible = this.state.visible.includes(item.id) ? true : false
		return (
			<Fragment>
				<TouchableOpacity onPress={() => this.lessonPress(item)}>
					<Notification
						style={styles.notification}
						key={`lesson${item.id}`}
						basic={
							<Text style={styles.basic}>
								{this.props.user.my_teacher.user.name.slice(
									0,
									NAME_LENGTH
								)}{" "}
								{strings("notifications.teacher_scheduled")}
							</Text>
						}
						leftSide={
							<View>
								<Text style={styles.lessonDate}>
									{moment
										.utc(item.date)
										.local()
										.format("DD.MM")}
								</Text>
								<Text style={styles.lessonHour}>
									<Hours
										duration={item.duration}
										date={item.date}
									/>
								</Text>
							</View>
						}
					/>
					<LessonPopup
						visible={visible}
						item={item}
						onPress={this.lessonPress}
						testID="lessonPopup"
						navigation={this.props.navigation}
					/>
				</TouchableOpacity>
			</Fragment>
		)
	}

	_renderItems = () => {
		if (this.state.loading) {
			return <StudentsLoader />
		}
		let render = this.renderLesson
		let type = "lesson"
		if (this.state.filter.includes("payments")) {
			render = this.renderPayment
			type = "payment"
		}
		return (
			<FlatList
				data={this.state.items}
				renderItem={render}
				onEndReached={this.endReached.bind(this)}
				keyExtractor={item => `${type}${item.id}`}
				ListEmptyComponent={this._renderEmpty}
				extraData={this.state.visible}
				onRefresh={() => this.onRefresh()}
				refreshing={this.state.refreshing}
			/>
		)
	}

	endReached = () => {
		if (!this.state.nextUrl) return
		this.setState(
			{
				page: this.state.page + 1,
				nextUrl: ""
			},
			() => {
				this._getItems(true)
			}
		)
	}

	_dropdownChange = (value, index, data) => {
		this.setState(
			{
				page: 1,
				nextUrl: "",
				filter: value
			},
			() => {
				this._getItems()
			}
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="notifications"
			text={strings("empty_notifications")}
			style={styles.empty}
		/>
	)

	render() {
		return (
			<View style={styles.container}>
				<View testID="NotificationsView" style={styles.notifications}>
					<PageTitle
						style={styles.title}
						title={strings("tabs.notifications_title")}
						leftSide={
							<View>
								<Dropdown
									containerStyle={styles.dropdown}
									value={this.state.filter}
									data={this.filterOptions}
									onChangeText={this._dropdownChange}
									dropdownMargins={{ min: 0, max: 20 }}
									dropdownOffset={{ top: 0, left: 0 }}
									pickerStyle={{ marginTop: 60 }}
								/>
							</View>
						}
					/>
					<View style={styles.items}>{this._renderItems()}</View>
				</View>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	notifications: {
		flex: 1,
		paddingRight: MAIN_PADDING,
		paddingLeft: MAIN_PADDING,
		marginTop: 20
	},
	items: {
		marginTop: 20
	},
	notification: {},
	title: { marginBottom: 0 },
	dropdown: {
		alignSelf: "flex-end",
		width: 160,
		marginTop: 4
	},
	empty: {
		marginTop: 100
	},
	amount: {
		marginTop: 12,
		fontSize: 20,
		fontWeight: "bold"
	},
	basic: {
		marginTop: 10,
		fontSize: 18
	},
	lessonDate: {
		fontSize: 22,
		fontFamily: "Assistant-Light"
	},
	lessonHour: {
		fontWeight: "bold",
		fontSize: 14,
		color: "rgb(12, 116, 244)"
	}
})

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService
	}
}

export default connect(mapStateToProps)(Notifications)
