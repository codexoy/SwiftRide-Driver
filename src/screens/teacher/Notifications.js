import React, { Fragment } from "react"
import {
	View,
	Text,
	TouchableHighlight,
	StyleSheet,
	FlatList,
	TouchableOpacity,
	Alert
} from "react-native"
import { connect } from "react-redux"
import { strings, errors } from "../../i18n"
import PageTitle from "../../components/PageTitle"
import { MAIN_PADDING, colors } from "../../consts"
import { Dropdown } from "react-native-material-dropdown"
import EmptyState from "../../components/EmptyState"
import StudentsLoader from "../../components/StudentsLoader"
import Notification from "../../components/Notification"
import NotificationButtons from "./NotificationButtons"
import Hours from "../../components/Hours"
import moment from "moment"
import LessonPopup from "../../components/LessonPopup"
import { fetchOrError, popLatestError } from "../../actions/utils"
import { API_ERROR } from "../../reducers/consts"
import ShowReceipt from "../../components/ShowReceipt"
import { Icon } from "react-native-elements"

export class Notifications extends React.Component {
	static navigationOptions = () => {
		return {
			title: "notifications",
			tabBarLabel: strings("tabs.notifications_title"),
			tabBarAccessibilityLabel: strings("tabs.notifications_title"),
			tabBarTestID: "NotificationsTab"
		}
	}
	constructor(props) {
		super(props)
		this.filterOptions = [
			{ value: "lessons/", label: strings("notifications.lessons") },
			{
				value: "teacher/students",
				label: strings("notifications.students")
			},
			{
				value: "lessons/payments",
				label: strings("notifications.payments")
			}
		]
		this.state = {
			items: [],
			page: 1,
			nextUrl: "",
			refreshing: false,
			visible: [],
			filter: this.filterOptions[0]["value"],
			loading: true,
			extraFilter: ""
		}

		if (!this.props.navigation.getParam("filter")) this._getItems()

		this._dropdownChange = this._dropdownChange.bind(this)
	}

	_onNavigationFocus = () => {
		if (this.props.navigation.getParam("filter")) {
			this.setState(
				{
					extraFilter:
						this.props.navigation.getParam("extraFilter") || "",
					filter: this.props.navigation.getParam("filter"),
					loading: true,
					page: 1,
					nextUrl: ""
				},
				() => {
					this._getItems()
					this.props.navigation.state.params.filter = "" // reset
				}
			)
		}
	}

	componentDidMount() {
		this.willFocusSubscription = this.props.navigation.addListener(
			"willFocus",
			payload => {
				this._onNavigationFocus()
			}
		)
	}

	componentWillUnmount() {
		this.willFocusSubscription.remove()
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			Alert.alert(strings("errors.title"), errors(error))
		}
	}

	_getItems = async (append = false) => {
		const resp = await this.props.fetchService.fetch(
			"/" +
				this.state.filter +
				"?limit=10&is_approved=false&order_by=created_at desc&page=" +
				this.state.page +
				this.state.extraFilter,
			{
				method: "GET"
			}
		)
		let newValue = resp.json.data
		if (append) {
			newValue = [...this.state.items, ...newValue]
		}
		this.setState({
			items: newValue,
			nextUrl: resp.json["next_url"],
			loading: false,
			refreshing: false
		})
	}

	onRefresh = () => {
		this.setState({ page: 1, nextUrl: "", refreshing: true }, () => {
			this._getItems()
		})
	}

	lessonPress = item => {
		let newVisible
		if (this.state.visible.includes(item.id)) {
			// we pop it
			newVisible = this.state.visible.filter((v, i) => v != item.id)
		} else {
			newVisible = [...this.state.visible, item.id]
		}
		this.setState({ visible: newVisible })
	}

	approve = async (type, item, index) => {
		const id = item.id || item.student_id
		const resp = await this.props.dispatch(
			fetchOrError(`/${type}/${id}/approve`, {
				method: "GET"
			})
		)
		if (resp) {
			Alert.alert(strings(`teacher.notifications.${type}_approved`))
			let newItems = [...this.state.items]
			newItems.splice(index, 1)
			this.setState({ items: newItems })
		}
	}

	delete = async (type, item, index) => {
		const id = item.id || item.student_id
		const resp = await this.props.dispatch(
			fetchOrError(`/${type}/${id}`, {
				method: "DELETE"
			})
		)
		if (resp) {
			Alert.alert(strings(`teacher.notifications.${type}_deleted`))
			let newItems = [...this.state.items]
			newItems.splice(index, 1)
			this.setState({ items: newItems })
		}
	}

	renderLesson = ({ item, index }) => {
		const visible = this.state.visible.includes(item.id) ? true : false
		return (
			<Fragment>
				<TouchableOpacity onPress={() => this.lessonPress(item)}>
					<Notification
						style={styles.notification}
						key={`lesson${item.id}`}
						user={item.student.user}
						name={`${item.student.user.name}(${
							item.lesson_number
						})`}
						type="new_lesson"
						leftSide={
							<View>
								<Text style={styles.date}>
									{moment
										.utc(item.date)
										.local()
										.format("DD.MM")}
								</Text>
								<Text style={styles.hour}>
									<Hours
										duration={item.duration}
										date={item.date}
									/>
								</Text>
							</View>
						}
					>
						<NotificationButtons
							approve={() => this.approve("lessons", item, index)}
							edit={() =>
								this.props.navigation.navigate("Lesson", {
									lesson: item
								})
							}
							delete={() => this.delete("lessons", item, index)}
						/>
					</Notification>
					<LessonPopup
						visible={visible}
						item={item}
						onPress={this.lessonPress}
						testID="lessonPopup"
						navigation={this.props.navigation}
					/>
				</TouchableOpacity>
			</Fragment>
		)
	}

	renderStudent = ({ item, index }) => {
		return (
			<Notification
				style={styles.notification}
				key={`student${item.student_id}`}
				name={`${item.user.name}`}
				user={item.user}
				type="new_student"
			>
				<NotificationButtons
					approve={() => this.approve("student", item, index)}
					delete={() => this.delete("student", item, index)}
				/>
			</Notification>
		)
	}

	renderPayment = ({ item, index }) => {
		return (
			<TouchableOpacity
				onPress={() =>
					this.props.navigation.navigate("StudentProfile", {
						student: item.student
					})
				}
			>
				<Notification
					style={styles.notification}
					key={`payment${item.id}`}
					leftSide={<Text style={styles.amount}>{item.amount}â‚ª</Text>}
					name={`${item.student.user.name}`}
					user={item.student.user}
					type="new_payment"
					extra={
						<ShowReceipt
							item={item}
							dispatch={this.props.dispatch}
							user={this.props.user}
						/>
					}
				/>
			</TouchableOpacity>
		)
	}

	_renderItems = () => {
		if (this.state.loading) {
			return <StudentsLoader />
		}
		let render = this.renderLesson
		let type = "lesson"
		let key = "id"
		if (this.state.filter.includes("students")) {
			render = this.renderStudent
			type = "student"
			key = "student_id"
		} else if (this.state.filter.includes("payments")) {
			render = this.renderPayment
			type = "payment"
		}
		return (
			<FlatList
				data={this.state.items}
				renderItem={render}
				onEndReached={this.endReached.bind(this)}
				keyExtractor={item => `${type}${item[key]}`}
				ListEmptyComponent={this._renderEmpty}
				extraData={this.state.visible}
				onRefresh={() => this.onRefresh()}
				refreshing={this.state.refreshing}
			/>
		)
	}

	endReached = () => {
		if (!this.state.nextUrl) return
		this.setState(
			{
				page: this.state.page + 1,
				nextUrl: ""
			},
			() => {
				this._getItems(true)
			}
		)
	}

	_dropdownChange = (value, index, data) => {
		this.setState(
			{
				items: [],
				page: 1,
				nextUrl: "",
				loading: true,
				filter: value,
				extraFilter: ""
			},
			() => {
				this._getItems()
			}
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="notifications"
			text={strings("empty_notifications")}
			style={styles.empty}
		/>
	)

	_removeExtraFilter = () => {
		this.setState({ extraFilter: "" }, () => {
			this._dropdownChange(this.state.filter)
		})
	}

	render() {
		let headerLeft,
			backButton,
			headerStyle = {},
			title,
			titleStyle = {}
		if (this.state.extraFilter != "") {
			backButton = (
				<TouchableOpacity
					onPress={() => this.props.navigation.goBack()}
				>
					<Icon name="arrow-forward" type="material" />
				</TouchableOpacity>
			)
			title = strings("student_profile.payments", {
				text: this.props.navigation.getParam("filterText")
			})
			headerStyle = styles.headerRow
			titleStyle = styles.title
		} else {
			headerLeft = (
				<View>
					<Dropdown
						containerStyle={styles.dropdown}
						value={this.state.filter}
						data={this.filterOptions}
						onChangeText={this._dropdownChange}
						dropdownMargins={{ min: 0, max: 20 }}
						dropdownOffset={{ top: 0, left: 0 }}
						pickerStyle={{ marginTop: 60 }}
					/>
				</View>
			)
		}
		return (
			<View style={styles.container}>
				<View testID="NotificationsView" style={styles.notifications}>
					<View style={headerStyle}>
						{backButton}
						<PageTitle
							style={titleStyle}
							title={title || strings("tabs.notifications_title")}
							leftSide={headerLeft}
						/>
					</View>

					<View style={styles.items}>{this._renderItems()}</View>
				</View>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	headerRow: {
		flexDirection: "row",
		justifyContent: "flex-start"
	},
	notifications: {
		flex: 1,
		paddingRight: MAIN_PADDING,
		paddingLeft: MAIN_PADDING,
		marginTop: 20
	},
	items: {
		marginTop: 20
	},
	notification: {},
	dropdown: {
		alignSelf: "flex-end",
		width: 160,
		marginTop: 4
	},
	empty: {
		marginTop: 100
	},
	date: {
		fontSize: 22,
		fontFamily: "Assistant-Light"
	},
	hour: {
		fontWeight: "bold",
		fontSize: 14,
		color: "rgb(12, 116, 244)"
	},
	amount: {
		marginTop: 16,
		fontSize: 20,
		fontWeight: "bold"
	},
	extraFilter: {
		flexDirection: "row",
		marginTop: -8
	},
	filterText: {
		fontWeight: "bold"
	},
	deleteFilterText: {
		color: colors.blue
	},
	title: {
		marginLeft: 8,
		marginTop: -4
	}
})

function mapStateToProps(state) {
	return {
		fetchService: state.fetchService,
		user: state.user,
		errors: state.errors
	}
}

export default connect(mapStateToProps)(Notifications)
