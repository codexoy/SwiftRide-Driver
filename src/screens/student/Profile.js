import React from "react"
import { connect } from "react-redux"
import { strings } from "../../i18n"
import StudentProfile from "../../components/StudentProfile"
import EditStudent from "../../components/EditStudent"
import Notifications from "../teacher/Notifications"
import { createStackNavigator } from "react-navigation"
import Topics from "../Topics"

export class Profile extends React.Component {
	static navigationOptions = () => {
		return {
			title: "notifications",
			tabBarLabel: strings("tabs.notifications_title"),
			tabBarAccessibilityLabel: strings("tabs.notifications_title"),
			tabBarTestID: "NotificationsTab"
		}
	}
	constructor(props) {
		// only here for the test suite to work
		super(props)
	}

	render() {
		return <StudentProfile {...this.props} />
	}
}

function mapStateToProps(state) {
	return {
		user: state.user,
		fetchService: state.fetchService,
		error: state.error
	}
}
const profileScreen = connect(mapStateToProps)(Profile)
const editScreen = connect(mapStateToProps)(EditStudent)
const profile = createStackNavigator(
	{
		Main: profileScreen,
		Topics,
		EditStudent: editScreen,
		Payments: Notifications
	},
	{
		initialRouteName: "Main",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
export default profile
