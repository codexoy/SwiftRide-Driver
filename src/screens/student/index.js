import {
	createBottomTabNavigator,
	createStackNavigator
} from "react-navigation"
import { tabBarOptions } from "../consts"
import Home from "./Home"
import Notifications from "./Notifications"
import Lesson from "./Lesson"
import Schedule from "./Schedule"
import Profile from "./Profile"
import Settings from "../Settings"
import StudentProfile from "../student/Profile"

export default createBottomTabNavigator(
	{
		Home: createStackNavigator(
			{
				Main: createStackNavigator(
					{ Home, Lesson },
					{
						initialRouteKey: "Home",
						headerMode: "none"
					}
				),
				Settings
			},
			{
				initialRouteName: "Main",
				mode: "modal",
				headerMode: "none",
				navigationOptions: {
					headerVisible: false
				}
			}
		),
		Notifications: createStackNavigator(
			{ Main: Notifications, Lesson },
			{
				initialRouteKey: "Main",
				headerMode: "none"
			}
		),
		AddLesson: Lesson,
		Schedule: createStackNavigator(
			{ Main: Schedule, Lesson },
			{
				initialRouteKey: "Main",
				headerMode: "none"
			}
		),
		StudentProfile: Profile
	},
	tabBarOptions
)
