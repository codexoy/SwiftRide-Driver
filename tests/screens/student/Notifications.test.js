import "react-native"
import React from "react"

import renderer from "react-test-renderer"

import { Notifications } from "../../../src/screens/student/Notifications"

describe("Notifications", () => {
	test("view renders correctly", done => {
		const student = { name: "test", student_id: 1 }
		fetch.mockResponseSuccess(
			JSON.stringify({
				data: [
					{
						duration: 40,
						student
					},
					{
						duration: 100,
						student
					}
				]
			})
		)
		const wrapper = shallow(
			<Notifications
				navigation={{
					...navigation,
					getParam: param => {
						return null
					}
				}}
				dispatch={dispatch}
				fetchService={fetchService}
				user={{
					my_teacher: { user: { id: 2 }, name: "teacher" },
					name: "student"
				}}
			/>
		)
		testAsyncComponent(wrapper, done)
	})
})
