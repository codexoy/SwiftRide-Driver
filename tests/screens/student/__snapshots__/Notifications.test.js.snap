// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
        "paddingLeft": 26,
        "paddingRight": 26,
      }
    }
    testID="NotificationsView"
  >
    <PageTitle
      leftSide={
        <View>
          <Dropdown
            absoluteRTLLayout={false}
            animationDuration={225}
            baseColor="rgba(0, 0, 0, .38)"
            containerStyle={
              Object {
                "alignSelf": "flex-end",
                "marginTop": 4,
                "width": 160,
              }
            }
            data={
              Array [
                Object {
                  "label": "שיעורים שנקבעו",
                  "value": "appointments/",
                },
                Object {
                  "label": "פעילויות כספיות",
                  "value": "appointments/payments",
                },
              ]
            }
            disabled={false}
            dropdownMargins={
              Object {
                "max": 20,
                "min": 0,
              }
            }
            dropdownOffset={
              Object {
                "left": 0,
                "top": 0,
              }
            }
            fontSize={16}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 4,
                "right": 4,
                "top": 6,
              }
            }
            itemColor="rgba(0, 0, 0, .54)"
            itemCount={4}
            itemPadding={8}
            labelExtractor={[Function]}
            onChangeText={[Function]}
            pickerStyle={
              Object {
                "marginTop": 60,
              }
            }
            propsExtractor={[Function]}
            rippleCentered={false}
            rippleDuration={400}
            rippleInsets={
              Object {
                "bottom": -8,
                "left": 0,
                "right": 0,
                "top": 16,
              }
            }
            rippleOpacity={0.54}
            rippleSequential={true}
            shadeOpacity={0.12}
            supportedOrientations={
              Array [
                "portrait",
                "portrait-upside-down",
                "landscape",
                "landscape-left",
                "landscape-right",
              ]
            }
            textColor="rgba(0, 0, 0, .87)"
            useNativeDriver={false}
            value="appointments/"
            valueExtractor={[Function]}
          />
        </View>
      }
      style={
        Object {
          "marginBottom": 0,
        }
      }
      title="פעילויות"
    />
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <FlatList
        ListEmptyComponent={[Function]}
        data={
          Array [
            Object {
              "duration": 40,
              "student": Object {
                "name": "test",
                "student_id": 1,
              },
            },
            Object {
              "duration": 100,
              "student": Object {
                "name": "test",
                "student_id": 1,
              },
            },
          ]
        }
        disableVirtualization={false}
        extraData={Array []}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReached={[Function]}
        onEndReachedThreshold={2}
        onRefresh={[Function]}
        refreshing={false}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>
  </View>
</View>
`;
